src/main/kotlin/org/veupathdb/lib/rabbit/jobs/model/ErrorNotification.kt

```diff
@@ -46,21 +50,27 @@ data class ErrorNotification @JvmOverloads constructor(
         throw IllegalStateException("Error notification has no ${JsonKey.JobID} field!")
       if (!json.has(JsonKey.Code))
         throw IllegalStateException("Error notification has no ${JsonKey.Code} field!")
+      if (!json.has(JsonKey.Body))
+        throw IllegalStateException("Error notification has no ${JsonKey.Body} field!")
 
       if (json.get(JsonKey.JobID).isNull)
         throw IllegalStateException("Error notification has a null ${JsonKey.JobID} field!")
       if (json.get(JsonKey.Code).isNull)
         throw IllegalStateException("Error notification has a null ${JsonKey.Code} field!")
+      if (json.get(JsonKey.Body).isNull)
+        throw IllegalStateException("Error notification has a null ${JsonKey.Body} field!")
 
       if (!json.get(JsonKey.JobID).isTextual)
         throw IllegalStateException("Error notification has a non-textual ${JsonKey.JobID} field!")
       if (!json.get(JsonKey.Code).isIntegralNumber)
         throw IllegalStateException("Error notification has a non-integral ${JsonKey.Code} field!")
 
       return ErrorNotification(
-        HashID(json.get(JsonKey.JobID).textValue()),
-        json.get(JsonKey.Code).intValue(),
-        json.get(JsonKey.Message).textValue()
+        jobID = HashID(json.get(JsonKey.JobID).textValue()),
+        code = json.get(JsonKey.Code).intValue(),
+        message = json.get(JsonKey.Message).textValue(),
+        attemptCount = json.get(JsonKey.AttemptCount).intValue(),
```

Comment:
With no absence check on `get(JsonKey.AttemptCount)` couldn't this throw an NPE?

AFAIK `ObjectNode.get` will return null if the key is absent and will return a `NullNode` if the key is explicitly set to `null`

Reply:
Thanks,fixed in new revision